app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,342 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.list
+
+import android.content.res.Configuration
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.defaultMinSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Favicon
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+
+val LIST_ITEM_HEIGHT = 56.dp
+
+val ICON_SIZE = 24.dp
+
+/**
+ * List item Composable used to display text with an optional second line of text and
+ * an optional interactable icon at the end.
+ *
+ * @param label The primary text in the item.
+ * @param description An optional second line of text.
+ * @param onClick Optional lambda for handling clicks.
```

Comment:
```suggestion
 * @param onClick Called when the user clicks on the item.
```

I have mostly been emulating the kDoc conventions from the Jetpack Compose components. I am mostly looking for what actions and its subject (ex, the list item) that will invoke the callback. 

<img width="719" alt="Screen Shot 2022-04-14 at 11 24 06 AM" src="https://user-images.githubusercontent.com/1190888/163422347-875d9382-16eb-49a4-b138-ea2676fff9e1.png">
<img width="621" alt="Screen Shot 2022-04-14 at 11 24 14 AM" src="https://user-images.githubusercontent.com/1190888/163422349-17be60ec-39af-4d6c-96a7-5bca9bea66bf.png">

Reply:
I think this makes sense.